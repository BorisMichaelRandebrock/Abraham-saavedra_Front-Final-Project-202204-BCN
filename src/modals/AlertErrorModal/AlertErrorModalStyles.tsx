import styled from "styled-components";

const AlertErrorModalSyles = styled.section`
  background-color: #b50000;
  width: 100vw;
  height: 70px;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 1;
  margin-top: 100px;
  p {
    color: #fff;
    font-size: medium;
    font-weight: 700;
  }

  animation: movein 1s ease forwards, moveout 1s 5s ease forwards;

  @keyframes movein {
    from {
      left: 200vw;
    }
    to {
      left: 0px;
    }
  }

  @keyframes moveout {
    from {
      left: 0px;
    }
    to {
      left: 200vw;
    }
  }
`;
export default AlertErrorModalSyles;
