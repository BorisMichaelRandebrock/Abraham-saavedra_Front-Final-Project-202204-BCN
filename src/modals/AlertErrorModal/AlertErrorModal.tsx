import AlertErrorModalStyles from "./AlertErrorModalStyles";

type TextModal = {
  text: string;
};

const AlertErrorModal = ({ text }: TextModal): JSX.Element => {
  return (
    <AlertErrorModalStyles>
      <p className="alert-modal">{text}</p>
    </AlertErrorModalStyles>
  );
};
export default AlertErrorModal;
