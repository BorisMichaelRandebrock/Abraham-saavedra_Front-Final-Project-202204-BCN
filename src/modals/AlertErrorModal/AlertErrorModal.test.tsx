import reactTestRenderer from "react-test-renderer";
import AlertErrorModal from "./AlertErrorModal";

describe("Given a AlertModal componet", () => {
  describe("When it's rendered", () => {
    test("Then it should show always the same as snapshot", () => {
      const textModal = "text for test";

      const modal = reactTestRenderer.create(
        <AlertErrorModal text={textModal} />
      );

      expect(modal).toMatchSnapshot();
    });
  });
});
