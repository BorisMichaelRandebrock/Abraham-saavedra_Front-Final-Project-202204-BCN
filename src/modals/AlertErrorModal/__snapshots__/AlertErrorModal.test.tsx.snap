// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a AlertModal componet When it's rendered Then it should show always the same as snapshot 1`] = `
<section
  className="sc-bczRLJ iOWcRp"
>
  <p
    className="alert-modal"
  >
    text for test
  </p>
</section>
`;
